doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -384,91 +381,104 @@ Rewriting with definitional equality
-   construction inside the :n:`@term` itself (use here the ``zeta`` flag). In
-   any cases, opaque constants are not unfolded (see :ref:`vernac-controlling-the-reduction-strategies`).
-
-   Normalization according to the flags is done by first evaluating the
-   head of the expression into a *weak-head* normal form, i.e. until the
-   evaluation is blocked by a variable (or an opaque constant, or an
-   axiom), as e.g. in :g:`x u1 ... un` , or :g:`match x with ... end`, or
-   :g:`(fix f x {struct x} := ...) x`, or is a constructed form (a
-   :math:`\lambda`-expression, a constructor, a cofixpoint, an inductive type, a
-   product type, a sort), or is a redex that the flags prevent to reduce. Once a
+.. tacn:: lazy {? @reductions } {? @occurrences }
+          cbv {? @reductions } {? @occurrences }
+
+   .. insertprodn reductions delta_reductions
+
+   .. prodn::
+      reductions ::= {+ @reduction }
+      | @delta_reductions
+      reduction ::= beta
+      | delta {? @delta_reductions }
+      | match
+      | fix
+      | cofix
+      | iota
+      | zeta
+      delta_reductions ::= {? - } [ {+ @reference } ]
+
+   Normalize the goal as specified by :n:`@reductions`.  If no reductions are
+   specified by name, all reductions are applied.  If any reductions are specified by name,
+   then only the named reductions are applied.  The reductions include:
+
+   `beta`
+     :term:`beta-reduction` of functional application
+
+   :n:`delta {? @delta_reductions }`
+     :term:`delta-reduction`: unfolding of transparent constants, see :ref:`controlling-the-reduction-strategies`.
+     The form in :n:`@reductions` without the keyword `delta` includes `beta`,
+     `iota` and `zeta` reductions in addition to `delta` using the given :n:`@delta_reductions`.
+
+     :n:`{? - } [ {+ @reference } ]`
+       without the `-`, limits delta unfolding to the listed constants.  If the
+       `-` is present,
+       unfolding is applied to all constants that are not listed.
+       Notice that the ``delta`` doesn't apply to variables bound by a let-in
+       construction inside the term itself (use ``zeta`` to unfold these).
+       Opaque constants are never unfolded except by :tacn:`vm_compute` and
+       :tacn:`native_compute`
+       (see https://github.com/coq/coq/issues/4476 and :ref:`controlling-the-reduction-strategies`).
+
+   `iota`
+     :term:`iota-reduction` of pattern matching (`match`) over a constructed term and unfolding
+     of :g:`fix` and :g:`cofix` expressions.  Shorthand for `match fix cofix`
+
+   `zeta`
+      :term:`zeta-reduction`: contraction of local definitions
+
+   If :n:`@reductions` is not specified, the conversions `beta delta iota zeta` are used.
+   If :n:`@reductions` is a list of :n:`@reduction`\s, then only the named conversions are used.
+   If :n:`@reductions` is a :n:`@delta_reductions`, then that conversion is applied
+   in addition to `beta iota zeta`.
```

Comment:
This is duplicative with the text above:
```
Normalize the goal as specified by :n:`@reductions`.  If no reductions are
   specified by name, all reductions are applied.  If any reductions are specified by name,
   then only the named reductions are applied.  The reductions include:
```
I'd suggest merging these two paragraphs.